### PR TITLE
escape echo variables using wordpress' native functions

### PR DESCRIPTION
**Pull request by cpsychogyios**

### CODE REVIEW

File: templates/setup.php [hunk 1]
```diff
@@ -1,35 +1,35 @@
 <div class="wrap">
 	<h1><?php _e('Setup', 'talentlms'); ?></h1>
 
-	<div id='action-message' class='<?php echo (isset($action_status)) ? $action_status : ''; ?> fade'>
-		<p><?php echo (isset($action_message)) ? $action_message : '' ?></p>
+	<div id='action-message' class='<?php echo (isset($action_status)) ? esc_html($action_status) : ''; ?> fade'>
```
Comment (epolikrati) [inline templates/setup.php]:
```suggestion
	<div id='action-message' class='<?php echo (isset($action_status)) ? esc_attr($action_status) : ''; ?> fade'>
```

File: templates/setup.php [hunk 2]
```diff
@@ -1,35 +1,35 @@
...
+		<p><?php echo (isset($action_message)) ? esc_html($action_message) : '' ?></p>
```
Comment (epolikrati) [inline templates/setup.php]:
```suggestion
		<p><?php echo (isset($action_message)) ? esc_html($action_message) : ''; ?></p>
```

File: templates/setup.php [hunk 3]
```diff
@@ -1,35 +1,35 @@
 <div class="wrap">
 	<h1><?php _e('Setup', 'talentlms'); ?></h1>
```
Comment (epolikrati) [inline templates/setup.php]:
```suggestion
	<h1><?php esc_html_e('Setup', 'talentlms'); ?></h1>
```

File: templates/setup.php [hunk 4]
```diff
@@ -1,35 +1,35 @@
...
 
-	<div id='action-message' class='<?php echo (isset($action_status)) ? $action_status : ''; ?> fade'>
-		<p><?php echo (isset($action_message)) ? $action_message : '' ?></p>
+	<div id='action-message' class='<?php echo (isset($action_status)) ? esc_html($action_status) : ''; ?> fade'>
+		<p><?php echo (isset($action_message)) ? esc_html($action_message) : '' ?></p>
 	</div>
 
 	<form name="talentlms-setup-form" method="post" action="<?php echo admin_url('admin.php?page=talentlms-setup'); ?>">
```
Comment (epolikrati) [inline templates/setup.php]:
```suggestion
	<form name="talentlms-setup-form" method="post" action="<?php echo esc_url(admin_url('admin.php?page=talentlms-setup')); ?>">
```

File: templates/setup.php [hunk 5]
```diff
@@ -1,35 +1,35 @@
...
 		<input type="hidden" name="action" value="tlms-setup">
 
 		<table class="form-table">
 			<tr>
-				<th scope="row" class="form-field form-required <?php echo $domain_validation; ?>">
+				<th scope="row" class="form-field form-required <?php echo esc_attr($domain_validation); ?>">
 					<label for="tlms-domain"><?php _e("TalentLMS Domain", 'talentlms'); ?> <span class="description">(<?php _e("Required", 'talentlms'); ?>)</span>:</label>
```
Comment (epolikrati) [inline templates/setup.php]:
```suggestion
					<label for="tlms-domain"><?php esc_html_e("TalentLMS Domain", 'talentlms'); ?> <span class="description">(<?php esc_html_e("Required", 'talentlms'); ?>)</span>:</label>
```

File: templates/setup.php [hunk 6]
```diff
@@ -1,35 +1,35 @@
...
 				</th>
-				<td class="form-field form-required <?php echo $domain_validation; ?>">
+				<td class="form-field form-required <?php echo esc_attr($domain_validation); ?>">
 					<input id="tlms-domain" name="tlms-domain" style="width: 25em;" value="<?php echo get_option('tlms-domain'); ?>"/>
 				</td>
 			</tr>
 			<tr>
-				<th scope="row" class="form-field form-required <?php echo $api_validation; ?>">
+				<th scope="row" class="form-field form-required <?php echo esc_attr($api_validation); ?>">
 					<label for="tlms-apikey"><?php _e("API Key", 'talentlms'); ?> <span class="description"><?php _e("(Required)", 'talentlms'); ?></span>:</label>
```
Comment (epolikrati) [inline templates/setup.php]:
```suggestion
					<label for="tlms-apikey"><?php esc_html_e("API Key", 'talentlms'); ?> <span class="description"><?php esc_html_e("Required", 'talentlms'); ?></span>:</label>
```

File: templates/setup.php [hunk 7]
```diff
@@ -1,35 +1,35 @@
...
 				</th>
-				<td class="form-field form-required <?php echo $api_validation; ?>">
+				<td class="form-field form-required <?php echo esc_attr($api_validation); ?>">
 					<input id="tlms-apikey" name="tlms-apikey" style="width: 25em;" value="<?php echo get_option('tlms-apikey'); ?>"/>
 				</td>
 			</tr>
 			<tr style="border-top: 1px dashed #c9c9c9">
-				<th scope="row" class="form-field form-required <?php echo $enroll_user_validation; ?>">
+				<th scope="row" class="form-field form-required <?php echo esc_attr($enroll_user_validation); ?>">
 					<label for="tlms-enroll-user-to-courses"><?php _e("Enroll user to courses", 'talentlms'); ?> <span class="description"><?php _e("(Required)", 'talentlms'); ?></span>:</label>
```
Comment (epolikrati) [inline templates/setup.php]:
```suggestion
					<label for="tlms-enroll-user-to-courses"><?php esc_html_e("Enroll user to courses", 'talentlms'); ?> <span class="description"><?php esc_html_e("(Required)", 'talentlms'); ?></span>:</label>
```

File: templates/setup.php [hunk 8]
```diff
@@ -1,35 +1,35 @@
...
 				</th>
-				<td class="form-field form-required <?php echo $enroll_user_validation; ?>">
+				<td class="form-field form-required <?php echo esc_attr($enroll_user_validation); ?>">
 					<select name="tlms-enroll-user-to-courses">
 						<option value="submission" <?php echo (get_option('tlms-enroll-user-to-courses') == 'submission')? 'selected="Selected"': ''; ?> ><?php _e("Upon order submission", 'talentlms'); ?></option>
```
Comment (epolikrati) [inline templates/setup.php]:
```suggestion
						<option value="submission" <?php echo (get_option('tlms-enroll-user-to-courses') == 'submission')? 'selected="Selected"': ''; ?> ><?php esc_html_e("Upon order submission", 'talentlms'); ?></option>
```

Comment (epolikrati) [inline templates/setup.php]:
```suggestion
						<option value="submission" <?php echo (get_option('tlms-enroll-user-to-courses') == 'submission')? 'selected="Selected"': ''; ?> ><?php esc_html_e("Upon order submission", 'talentlms'); ?></option>
```

File: templates/setup.php [hunk 9]
```diff
@@ -1,35 +1,35 @@
...
 						<option value="completion" <?php echo (get_option('tlms-enroll-user-to-courses') == 'completion')? 'selected="Selected"': ''; ?> ><?php _e("Upon order completion", 'talentlms'); ?></option>
```
Comment (epolikrati) [inline templates/setup.php]:
```suggestion
						<option value="completion" <?php echo (get_option('tlms-enroll-user-to-courses') == 'completion')? 'selected="Selected"': ''; ?> ><?php esc_html_e("Upon order completion", 'talentlms'); ?></option>
```

**Review by epolikrati — CHANGES_REQUESTED** [review]

**Review by epolikrati — APPROVED** [review]

**Review by epolikrati — APPROVED** [review]
@cpsychogyios There are lot of places in our code base in which the data must be escaped. After running static analysis and cs tools, they'll be automatically replaced & fixed.

**Comment by cpsychogyios** [comment]
> @cpsychogyios There are lot of places in our code base in which the data must be escaped. After running static analysis and cs tool, they'll be automatically replaced.

@epolikrati We don't know that yet, first we have to run them and then evaluate what they will suggest us. Definitely many of these changes would be reported by these tools as well.

**Pull request merged**